src/ds/d400/d400-options.cpp

```diff
@@ -466,4 +466,66 @@ namespace librealsense
     {
         snapshot = std::make_shared<const_value_option>(get_description(), 0.f);
     }
-}
+
+    void librealsense::gyro_sensitivity_option::set( float value ) 
+    {
+        auto strong = _sensor.lock();
+        if( ! strong )
+            throw invalid_value_exception( "Hid sensor is not alive for setting" );
+
+        if( strong->is_streaming() )
+            throw invalid_value_exception( "setting this option during streaming is not allowed!" );
+
+        if(!is_valid(value))
+            throw invalid_value_exception( "set(gyro_sensitivity) failed! Invalid Gyro sensitivity resolution request "
+                                           + std::to_string( value ) );
+
+        _value = value;
+        strong->set_imu_sensitivity( RS2_STREAM_GYRO, value );
+    }
+
+    float librealsense::gyro_sensitivity_option::query() const
+    {
+        return _value;
+    }
+
+    
+    const char * librealsense::gyro_sensitivity_option::get_value_description( float val ) const
+    {
+        switch( static_cast< int >( val ) )
+        {
+            case RS2_GYRO_SENSITIVITY_61_0_MILLI_DEG_SEC: {
+                return "61.0";
```

Comment:
Please be more verbose - "61.0 mDeg/Sec sensitivity"

Reply:
This can be done in the to-string.cpp
We have special cases there when we override the enum automatic conversion to our own string.
It will be weird adding sensitivity word again to the enum (and it might make the combo box string to long to show)
Take a look on special cases on to-string

Reply:
changed it to resolution